src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
+    private static class Node<T> {
+        T item;
+        Node<T> prev;
+        Node<T> next;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            head = new Node<>(null, value, null);
+            tail = head;
+            size++;
+            return true;
+        }
+        Node<T> newNode = new Node<>(tail, value, null);
+        tail.next = newNode;
+        tail = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            checkingIndex(index);
+        }
+        Node<T> currentNode = head;
+        for (int i = 0; i <= size; i++) {
+            if (index == size || index == 0 && size == 0) {
+                add(value);
+                return;
+            }
+            if (i == index) {
+                Node<T> newNode = new Node<>(currentNode.prev, currentNode.item, currentNode.next);
+                currentNode.item = value;
+                currentNode.prev = newNode.prev;
+                currentNode.next = newNode;
+                newNode.prev = currentNode;
+                size++;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
 
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
+        checkingIndex(index);
+        Node<T> currentNode = head;
+        for (int i = 0; i <= size; i++) {
+            if (i == index) {
+                return currentNode.item;
+            }
+            currentNode = currentNode.next;
+        }
```

Comment:
let's move this logic to the separate method: getNodeByIndex() and use it in get(), set(), remove(), add by index() methods